libbeat/autodiscover/providers/jolokia/discovery.go

```diff
@@ -0,0 +1,296 @@
+package jolokia
+
+import (
+	"encoding/json"
+	"net"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/bus"
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Jolokia Discovery query
+// {
+//   "type": "query"
+// }
+//
+// Example Jolokia Discovery response
+// {
+//   "agent_version": "1.5.0",
+//   "agent_id": "172.18.0.2-7-1322ae88-servlet",
+//   "server_product": "tomcat",
+//   "type": "response",
+//   "server_vendor": "Apache",
+//   "server_version": "7.0.86",
+//   "secured": false,
+//   "url": "http://172.18.0.2:8778/jolokia"
+// }
+//
+// Example discovery probe with socat
+//
+//   echo '{"type":"query"}' | sudo socat STDIO UDP4-DATAGRAM:239.192.48.84:24884,interface=br0 | jq .
+//
+
+// Message contains the information of a Jolokia Discovery message
+var messageSchema = s.Schema{
+	"agent": s.Object{
```

Comment:
So this is not the `agent` that ships the data but the jolokia agent that discovered the data. Just trying to get my head around this for ECS: https://github.com/elastic/ecs#agent The `agent` on the top level in these cases would still be metricbeat or filebeat, right?

Reply:
Yes, this is about the jolokia agent, under the `jolokia` namespace.